tests/cpp/aot/llvm/utils.py

```diff
@@ -121,3 +121,48 @@ def run(base: int, arr: ti.types.ndarray()):
     m = ti.aot.Module(arch)
     m.add_kernel(run, template_args={'arr': arr})
     m.save(dir_name, 'whatever')
+
+
+def compile_graph_aot(arch):
+    ti.init(arch=arch)
+
+    @ti.kernel
+    def run0(base: int, arr: ti.any_arr(field_dim=1)):
+        for i in arr:
+            arr[i] += [base + i]
+
+    @ti.kernel
+    def run1(base: int, arr: ti.any_arr(field_dim=1)):
+        for i in arr:
+            arr[i] += [base + i]
+
+    @ti.kernel
+    def run2(base: int, arr: ti.any_arr(field_dim=1)):
+        for i in arr:
+            arr[i] += [base + i]
+
+    arr = ti.graph.Arg(ti.graph.ArgKind.NDARRAY,
+                       'arr',
+                       ti.i32,
+                       element_shape=(1, ))
+
+    base0 = ti.graph.Arg(ti.graph.ArgKind.SCALAR, 'base0', ti.i32)
```

Comment:
FYI the latest master will require specifying `field_dim` here as well. Fixing this should be good enough to pass all tests after rebasing on top of master.